qbraid/runtime/azure/provider.py

```diff
@@ -0,0 +1,494 @@
+# Copyright (C) 2024 qBraid
+#
+# This file is part of the qBraid-SDK
+#
+# The qBraid-SDK is free software released under the GNU General Public License v3
+# or later. You can redistribute and/or modify it under the terms of the GPL v3.
+# See the LICENSE file in the project root or <https://www.gnu.org/licenses/gpl-3.0.html>.
+#
+# THERE IS NO WARRANTY for the qBraid-SDK, as per Section 15 of the GPL v3.
+
+"""
+Module defining Azure session and provider classes
+
+"""
+from typing import Any
+from uuid import uuid4
+
+import openqasm3
+from azure.storage.blob import BlobServiceClient, ContentSettings
+from qbraid_core._import import LazyLoader
+from qbraid_core.sessions import Session
+
+from qbraid.programs.spec import ProgramSpec
+from qbraid.runtime.enums import DeviceType
+from qbraid.runtime.profile import TargetProfile
+from qbraid.runtime.provider import QuantumProvider
+
+from .device import AzureQuantumDevice
+
+pyquil = LazyLoader("pyquil", globals(), "pyquil")
+
+
+class AzureSession(Session):
+    """Azure session class."""
+
+    def __init__(
+        self,
+        client_id: str,
+        client_secret: str,
+        tenant_id: str,
+        location_name: str,
+        subscription_id: str,
+        resource_group_name: str,
+        workspace_name: str,
+        storage_account: str,
+        connection_string: str,
+    ):  # pylint: disable= too-many-arguments
+
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.tenant_id = tenant_id
+        self.location_name = location_name
+        self.subscription_id = subscription_id
+        self.resource_group_name = resource_group_name
+        self.workspace_name = workspace_name
+        self.storage_account = storage_account
+        self.connection_string = connection_string
+        self.backends = {
+            "Syntax Checker": "quantinuum.sim.h1-1sc",
+            "Emulator": "quantinuum.sim.h1-1e",
+            "QPU": "quantinuum.qpu.h1-1",
+        }  # will need to add for pyquil
+        self.formats = {"quantinuum": ["honeywell.openqasm.v1", "honeywell.quantum-results.v1"]}
+
+        self.jobs = {}
+        self.session = Session()
+        self._helpers = AzureHelperFunctions()
+
+        self.base_url = (f"https://{location_name}.quantum.azure.com",)
+
+        super().__init__(base_url=self.base_url)
+
+    def get_devices(self) -> list[str, dict[str, Any]]:
```

Comment:
Return type of this function looks like a `list[dict[str, Any]]`, not a `list[str, dict[str, Any]]`